switch/switch_main.ml

```diff
@@ -176,9 +176,17 @@ let make_server config =
       >>= fun qs ->
       on_disk_queues := qs;
       info "Reading the redo-log from %s" redo_log_path;
-      let open Lwt_result in
-      Block.connect ("buffered:" ^ redo_log_path)
+      Block.connect ~buffered:true redo_log_path
+      >>= fun block -> begin
+        match block with
+        | `Ok block -> Lwt.return block
+        | `Error e ->
```

Comment:
I'm not too familiar with this code as I've only rebased it, but I like this suggestion. I think this is a trade-off between having less nesting and grouping error handling into one block while not repeating the `` `Error`` tag. At least we should put the nested `match` into a `begin`-`end` block to avoid branches accidentally getting into the wrong pattern match. Anyway, I'll just merge the two `match` blocks into one, because that's clearer in this case.

As for the `__LOC__`, I assume it will always say the line number where it appears, `"Other error in 186"`, so is this to make it easier to find out which source code line the error message is coming from?